lib/TPP/Dialect/Xsmm/XsmmOps.cpp

```diff
@@ -16,3 +16,82 @@
 
 using namespace mlir;
 using namespace mlir::xsmm;
+
+template <typename EnumClass>
+static ParseResult parseEnum(EnumClass &value, OpAsmParser &parser) {
+  StringRef flag;
+  auto loc = parser.getCurrentLocation();
+  if (parser.parseKeyword(&flag))
+    return failure();
+  auto flagAttr = symbolizeEnum<EnumClass>(flag);
+  if (!flagAttr)
+    return parser.emitError(loc, "invalid enum") << flag;
+  value = *flagAttr;
+  return success();
+}
+
+static ParseResult parserImpl(OpAsmParser &parser, OperationState &result) {
```

Comment:
Missing `parseOptionalAttrDict` for `attr-dict`

Reply:
Thanks, do we need it?

Reply:
Some generic passes could attach attributes. I think it's standard for all ops to support attr-dict.